### PR TITLE
Mender integration for Toradex Colibri-imx6ull SoM

### PR DESCRIPTION
**Pull request by rickyricksanchez**

### CODE REVIEW

**Comment by mrpelotazo** [comment]
@rickyricksanchez

* It seems that **teziimg** has to be removed from **IMAGE_FSTYPES**, same as for the **verdin-imx8mm** and **colibri-imx7** (zeus branch) machines since **do_image_teziimg** task fails otherwise.

* There is the following warning if **mtdimg** is not removed from **IMAGE_FSTYPES** too:

```
WARNING: fb-cops-gw-development-image-1.0-r0 do_image_mtdimg: Don't know how to flash mtdparts 'mx6ull-bcb'. Filling with zeros.
WARNING: fb-cops-gw-development-image-1.0-r0 do_image_mtdimg: Don't know how to flash mtdparts 'u-boot1'. Filling with zeros.
WARNING: fb-cops-gw-development-image-1.0-r0 do_image_mtdimg: Don't know how to flash mtdparts 'u-boot2'. Filling with zeros.
```

It is recommended by the [mender documentation](https://docs.mender.io/system-updates-yocto-project/board-integration/raw-flash#mender_mtdparts) to disable it for custom MENDER_MTDPARTS. I've tested it and the warning is gone for good.

Summing up, adding `IMAGE_FSTYPES_remove_colibri-imx6ull = " mtdimg teziimg"` to `local.conf.append.colibri-imx6ull` should be sufficient.

* Last thing I've noticed is that **MENDER_RESERVED_SPACE_BOOTLOADER_DATA** is placed within **llocal.conf.append.colibri-imx6ull**. I was wondering if it should be better placed in the **u-boot-toradex_%.bbappend** file since AFAIK its value is tied to the **BOOTENV_SIZE_mender-uboot_colibri-imx6ull** value defined in there: What do you think?

Great work btw!

**Comment by mirzak** [comment]
Thanks @rickyricksanchez.

And thanks for reviewing @mrpelotazo. Valid points, and agree with your suggestions.

**Comment by mrpelotazo** [comment]
@mirzak Is there something else to be done for this to get merged?

**Comment by mirzak** [comment]
I was waiting for the updates that you suggested in https://github.com/mendersoftware/meta-mender-community/pull/205#issuecomment-742430856. Can not see  that they have been resolved?

**Comment by mrpelotazo** [comment]
I see the PR is open and no further commits besides the original one. I'm I missing something?

**Comment by mirzak** [comment]
I was thinking @rickyricksanchez was working on it.

Do not think you can do changes to this PR directly unless you have write access to ` rickyricksanchez:dunfell`. If you want to take over you can create a new PR and we can close this one.

**Comment by mrpelotazo** [comment]
> I was thinking @rickyricksanchez was working on it.
Same here.

> If you want to take over you can create a new PR and we can close this one.
Will do.

**Comment by jsolla** [comment]
Sorry guys, been busy with other things and couldn't pay attention to this. Im back so I will prepare changes tomorrow morning.
@mrpelotazo I think you should be also added to the maintainers list. Your collaboration has been very productive for this to be done :)

**Comment by rickyricksanchez** [comment]
Sorry guys, I was busy with other things and didn't have the time to prepare this. Im free now so I will prepare the last change inmediately.

@mrpelotazo I think you should be added to the maintainers list, as your proposals were crucial for the success of the implementation.

**Comment by rickyricksanchez** [comment]
@mrpelotazo about the MENDER_RESERVED_SPACE_BOOTLOADER_DATA location. My understanding is safest place to assign it would be a config file since this way is global for the whole project rather than setting it on the bootloader recipe where it would be local to u-boot only. 

If my assumption is wrong, I would be more than happy to change it.

@mirzak What's your opinion on this?

**Comment by mrpelotazo** [comment]
> @mrpelotazo could you please provide an email address so I can add you to maintaners/contributors list?

mrpelotazo@posteo.net

**Comment by mrpelotazo** [comment]
> @mrpelotazo about the MENDER_RESERVED_SPACE_BOOTLOADER_DATA location: My understanding is safest place to assign it would be a config file since this way is global for the whole build rather than setting it on the bootloader recipe where it would be local to u-boot only.

I see your point there.

I guess my rationale is the other way around i.e. to keep local.conf as small as possible and not to add things in there preventively but when its necessary.

It looks to me that **MENDER_RESERVED_SPACE_BOOTLOADER_DATA** is only used for u-boot configuration, so it should not be necessary for it to be global: I tested it and seems to work just fine. Defined in the u-boot append file as 

**MENDER_RESERVED_SPACE_BOOTLOADER_DATA_mender-uboot_colibri-imx6ull ="0x40000"**

Additionally, as I mentioned before, its value is tied to the **BOOTENV_SIZE** defined in the u-boot append file. So, both values are dependent on each other. Exposing one of them to the local conf will probably lead to people to mess around with one value without being aware of the other and consequently breaking the build.

Would also like to see what @mirzak  has to say to this. Probably he has some more insight that we do.

**Comment by rickyricksanchez** [comment]
> I see your point there.
> 
> I guess my rationale is the other way around i.e. to keep local.conf as small as possible and not to add things in there preventively but when its necessary.
> 
> It looks to me that **MENDER_RESERVED_SPACE_BOOTLOADER_DATA** is only used for u-boot configuration, so it should not be necessary for it to be global: I tested it and seems to work just fine. Defined in the u-boot append file as
> 
I fully agree, the less in the conf file the better IMHO. So since you checked and I also reviewed it, I decided to move it to the u-boot-toradex recipe. 

Unless @mirzak has something else to comment, I would say this one is ready to go :)

**Comment by mrpelotazo** [comment]
@rickyricksanchez thanks. I've just tested it with the last changes with both tezi and mender client. Seems to be all good.

@mirzak from my side also this is ready to go.

**Comment by giobauermeister** [comment]
I also tested this on Colibri iMX6ULL and worked fine. Yocto integration and build, provisioning and all. Thanks for the work.

**Comment by rickyricksanchez** [comment]
Hi @mirzak 
Any chances this could be merged anytime soon? :)

**Comment by mirzak** [comment]
Thank you to everyone involved in this. I have been out-of-office for a while and apologies for any delays caused by this.
